### PR TITLE
agent: Remove unnecessary OCI process config passing

### PR DESCRIPTION
**Pull request by ManaSugi**
The Kata agent gets an OCI process config through IPC only when the process's `init` is `false`.
If the process's `init` is `true`, a child process in the agent just copies the OCI process inside the whole OCI spec.
This removes trivial overhead by IPC.

Fixes: #2286

Signed-off-by: Manabu Sugimoto <Manabu.Sugimoto@sony.com>

### CODE REVIEW

**Comment by ManaSugi** [comment]
rebased

**Comment by ManaSugi** [comment]
passed first tests.

**Review by liubin — DISMISSED** [review]

**Comment by jodh-intel** [comment]
Any further update on this @ManaSugi?

**Comment by jodh-intel** [comment]
Ping @ManaSugi - please can you look at the outstanding queries on this PR?

**Comment by ManaSugi** [comment]
@jodh-intel Sorry for the late update. After updating #2785, I'll update this PR too.

File: src/agent/rustjail/src/container.rs
Comment (ManaSugi) [reply src/agent/rustjail/src/container.rs]: I fixed it. Thank you.

File: src/agent/rustjail/src/container.rs
Comment (ManaSugi) [reply src/agent/rustjail/src/container.rs:1251]: @jodh-intel Regarding a unit test of these changes, I'ld like to add the test to the `integration/agent/agent_test.sh` was proposed by https://github.com/kata-containers/tests/pull/4341 because the code path of theses changes is included in `do_init_child()` function. I'll create the test after the script is merged into main.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

**Comment by jodh-intel** [comment]
Hi @ManaSugi - Since there is debate as to whether this approach will work, I think a new test would help convince us.

ooi...

- How did you find this issue?
- Is there a measurable performance improvement with this PR?

**Comment by arronwy** [comment]
Can one of the admins verify this patch?

**Comment by gkurz** [comment]
Ping @ManaSugi . Can you answer @jodh-intel 's questions in comment https://github.com/kata-containers/kata-containers/pull/2287#issuecomment-1078781453 ? Also it seems this PR has conflicts and needs rebase.

**Comment by fidencio** [comment]
@ManaSugi, we're closing this PR as it's been opened and inactive for more than one year.
If you think this is a mistake and the PR should be re-opened, please, feel free to do so.

**Pull request closed without merging**
